feel/feelfilters/loadMEDmesh.hpp

```diff
@@ -0,0 +1,137 @@
+/* -*- mode: c++; coding: utf-8; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; show-trailing-whitespace: t -*-
+
+  This file is part of the Feel library
+
+  Author(s): Christophe Prud'homme <christophe.prudhomme@feelpp.org>
+       Date: 2013-12-24
+
+  Copyright (C) 2013-2016 Feel++ Consortium
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+/**
+   \file loadmeshmesh.hpp
+   \author Christophe Trophime <christophe.trophime@lncmi.cnrs.fr>
+   \date 2016-11-30
+ */
+#if !defined(FEELPP_LOADMEDMESH_HPP)
+#define FEELPP_LOADMEDMESH_HPP 1
+
+#include <feel/feelfilters/gmsh.hpp>
+#include <feel/feelfilters/detail/mesh.hpp>
+#include <feel/feelfilters/importergmsh.hpp>
+
+namespace Feel {
+/**
+ *
+ * \brief load a mesh data structure (hold in a shared_ptr<>) using GMSH
+ *
+ * \arg mesh mesh data structure
+ * \arg filename filename string (with extension)
+ * \arg refine optionally refine with \p refine levels the mesh (default: 0)
+ * \arg update update the mesh data structure (build internal faces and edges) (default : true)
+ * \arg physical_are_elementary_regions boolean to load specific meshes formats (default : false)
+ */
+BOOST_PARAMETER_FUNCTION(
+    ( typename Feel::detail::mesh<Args>::ptrtype ), // return type
+    loadMEDMesh,    // 2. function name
```

Comment:
I agree that would be great. The only point is that MED is a specific case...
for other extension there is a correspondance between the extension and a readXXX method in Gmsh.

But from a practical point of view I've no idea how to code that (beside having some kind dictionnaries linking extension and pointer to the proper readXXX method )